### ISSUE TITLE
Стиль кода

### ISSUE THREAD

**Issue by Bogenbai** [OdareNNbI/Falling-box#4]
- В [Background.cs](https://github.com/OdareNNbI/Falling-box/blob/master/Project/FallingBox/Assets/Scripts/Background.cs) на строках 20 и 23 следует взять в скобки выражения, для более лучшей читаемости:
Как сейчас:
`if (spritesRenderers[i].transform.position.y - spritesRenderers[i].bounds.extents.y >=
CameraManager.Instance.CameraDownYPosition)`
Следует:
`if ((spritesRenderers[i].transform.position.y - spritesRenderers[i].bounds.extents.y) >=
CameraManager.Instance.CameraDownYPosition)`
Так же в этих же строках рекомендуется присваивать значения, полученные как `spritesRenderers[i].transform.position.y` отдельной переменной, чтобы укоротить выражение в if'e.

- В [Level.cs](https://github.com/OdareNNbI/Falling-box/blob/master/Project/FallingBox/Assets/Scripts/Level.cs) так же следует присвоить значение на 110 строке, полученное как `existedPlatforms[i].transform.position.y`, отдельной переменной. То же самое на 124 строке.

- В [Plaform.cs](https://github.com/OdareNNbI/Falling-box/blob/master/Project/FallingBox/Assets/Scripts/Platform.cs) на строках 64 и 71 так же следует взять в скобки выражения, для более лучшей читаемости:
Как сейчас:
`if (transform.position.x + SpriteHalfXSize >= cameraRightXPosition)`
Следует:
`if ((transform.position.x + SpriteHalfXSize) >= cameraRightXPosition)`

- В [BaseManager.cs](https://github.com/OdareNNbI/Falling-box/blob/master/Project/FallingBox/Assets/Scripts/BaseManager.cs) на строке 20 следует использовать сокращенную форму создания экземпляра для объекта с неявным типизированием.
Как сейчас:
`GameObject obj = new GameObject(typeof(T).ToString() + " Auto Singleton");`
Следует:
`var obj = new GameObject(typeof(T).ToString() + " Auto Singleton");`
Так же в этом же файле в строках 11-27 присутствуют лишние табуляции.

- В [Box.cs](https://github.com/OdareNNbI/Falling-box/blob/master/Project/FallingBox/Assets/Scripts/Box.cs) в методе OnCollisionEnter2D() присутствует лишний контекстный блок.

**Comment by OdareNNbI** [comment]
1 пункт - с первой частью согласен, со второй не совсем, так как эту переменную я использую один раз и кешировать ее нету смысла,то не улучшить производительность
2 пункт - тоже самое, что и по первому 
3 пунтк - не вижу смысла тут тоже ставить скобки, так как выражение не такое уж длинное, чтобы было его плохо прочитать, плюс тут оно всего лишь одно.
4 пункт - исправил 
5 пункт - исправил

**Comment by Bogenbai** [comment]
> 1 пункт - с первой частью согласен, со второй не совсем, так как эту переменную я использую один раз и кешировать ее нету смысла,то не улучшить производительность

Тут же вопрос не в улучшении производительности, а в более лучшей читаемости кода

> 3 пунтк - не вижу смысла тут тоже ставить скобки, так как выражение не такое уж длинное, чтобы было его плохо прочитать, плюс тут оно всего лишь одно.

А вот майкрософт считает, что еще как нужно ставить тут скобки. Пример со страницы код конвеншена: 

> Use parentheses to make clauses in an expression apparent, as shown in the following code.
`if ((val1 > val2) && (val1 > val3))
{
    // Take appropriate action.
}`

**Comment by OdareNNbI** [comment]
Да, если ты используешь более одного выражения в скобках, я согласен, но у меня оно одно. И еще в конвеншене берётся полностью выражение, а не часть его. В этом и суть упрощать читаемость для нескольких выражений, отделяя их скобками, а если оно одно то выделять его никак не надо

**Comment by Bogenbai** [comment]
Скорее всего да, так оно и есть, но мне почему-то кажется, что они имеют в виду именно разделение операндов
